Source/WebKit/UIProcess/Cocoa/WebProcessProxyCocoa.mm

```diff
@@ -367,6 +373,53 @@
     protectedThrottler()->setShouldTakeNearSuspendedAssertion(false);
 }
 
+RefPtr<XPCEventHandler> WebProcessProxy::xpcEventHandler() const
+{
+    return adoptRef(new WebProcessProxy::XPCEventHandler(*this));
+}
+
+bool WebProcessProxy::XPCEventHandler::handleXPCEvent(xpc_object_t event) const
+{
+#if ENABLE(LOGD_BLOCKING_IN_WEBCONTENT)
+    auto messageName = xpcDictionaryGetString(event, XPCEndpoint::xpcMessageNameKey);
+    if (messageName == logMessageName) {
+        RefPtr webProcess = m_webProcess.get();
+        if (!webProcess)
+            return true;
+        if (!webProcess->m_logEndpointEnabled)
+            return true;
+
+        auto subsystem = xpcDictionaryGetString(event, subsystemKey);
+        auto category = xpcDictionaryGetString(event, categoryKey);
+        auto messageString = xpcDictionaryGetString(event, messageStringKey);
+        auto logType = xpc_dictionary_get_uint64(event, logTypeKey);
+        auto pid = xpc_connection_get_pid(xpc_dictionary_get_remote_connection(event));
+
+        OSObjectPtr<os_log_t> osLog;
+        if (!subsystem.isEmpty() && !category.isEmpty())
+            osLog = adoptOSObject(os_log_create(subsystem.utf8().data(), category.utf8().data()));
+
+        auto osLogPointer = osLog ? osLog.get() : OS_LOG_DEFAULT;
+        os_log_with_type(osLogPointer, static_cast<os_log_type_t>(logType), "WebContent[%d] %{public}s", static_cast<int>(pid), messageString.utf8().data());
+        ++webProcess->m_logCountForTesting;
+    } else if (messageName == disableLogMessageName) {
```

Comment:
As in my previous review, I still do not understand why we need the WebProcess to send a `disableLog` message to the UIProcess. I don't think I got an answer to my question? Why can't the WebProcess just stop sending messages (it seems like it always does)?

Reply:
Is it really worth the extra complexity? Don't we anyway destroy both the xpc_connection and this object once the process is done launching?

Reply:
That is a good point! We destroy the XPC connection on iOS, so this is only relevant on macOS. I can conditionalize on that feature :)